compiler-core/src/ast/tests.rs

```diff
@@ -240,6 +240,14 @@ fn find_node_sequence() {
     assert!(block.find_node(7).is_none());
 }
 
+// The function exits early when attempting to find the AST node for a specific index,
+// if the remaining elements have indices beyond the search index, returning none.
```

Comment:
Could you remove these comments about returning early please, they are unrelated to these tests.